server/controllers/proxyController.js

```diff
@@ -16,29 +17,63 @@ const proxy = httpProxy.createProxyServer({
 proxy.on('proxyReq', function(proxyReq, req, res, options) {
   proxyReq.setHeader('authorization', `Bearer ${AIRTABLE_API_KEY}`);
 });
-
 proxy.on('proxyRes', function(proxyRes, req, res) {
+  var body = [];
   proxyRes.on('data', function(chunk) {
+    body.push(chunk);
+  });
+  proxyRes.on('end', function() {
+    const buffer = Buffer.concat(body);
+    let proxyPayload = null;
+
     if (proxyRes.headers[`${CONTENT_ENCODING}`] == `${GZIP}`) {
-      zlib.gunzip(chunk, function(err, dezipped) {
-        let unzipPayload = dezipped.toString();
-        cache.set(req, JSON.parse(unzipPayload));
-        return res.send(JSON.parse(unzipPayload));
+      zlib.gunzip(buffer, function(err, decoded) {
+        if (!err) {
+          proxyPayload = JSON.parse(decoded.toString());
+          if (req.method === 'GET') {
+            cache.set(req, proxyPayload);
+          }
+          res.status(proxyRes.statusCode).send(proxyPayload);
+          res.end();
+        }
+      });
+    } else if (proxyRes.headers[`${CONTENT_ENCODING}`] == `${DEFLATE}`) {
+      zlib.inflate(buffer, function(err, decoded) {
+        if (!err) {
+          proxyPayload = JSON.parse(decoded.toString());
+          if (req.method === 'GET') {
+            cache.set(req, proxyPayload);
+          }
+          res.status(proxyRes.statusCode).send(proxyPayload);
+          res.end();
+        }
       });
     } else {
-      res.status(proxyRes.statusCode).send(JSON.parse(chunk.toString()));
+      proxyPayload = JSON.parse(buffer.toString());
+      if (req.method === 'GET') {
+        cache.set(req, proxyPayload);
+      }
+      res.status(proxyRes.statusCode).send(proxyPayload);
+      res.end();
     }
   });
 });
 
 module.exports = {
   web(req, res) {
-    const content = cache.get(req);
-    if (content) {
-      return res.status(200).send(content);
+    if (req.method === 'GET') {
+      const content = cache.get(req);
+      if (content) {
+        return res.status(200).send(content);
+      }
+      proxy.web(req, res, {
+        target: `${AIRTABLE_ENDPOINT_URL}`
+      });
+    } else {
+      cache.clear(req);
```

Comment:
We might need to be careful about our caching strategy. I just thought of this scenario:

1. User requests access to https://api.airtable.com/v0/appXYZ/Users/recABC (Retrieve specific record - becomes cached as https://api.airtable.com/v0/appXYZ/Users/recABC)
2. User updates recABC with a PATCH to https://api.airtable.com/v0/appXYZ/Users (Update specific record(s), busts cache for https://api.airtable.com/v0/appXYZ/Users)
3. User performs Step 1 again and gets back cached result, _not_ their updated version.

At the very least, we should do some parsing to strip off any "recABC" bits from the URL that gets cached.

Another consideration is query strings like `filterByFormula` applied to GET requests: should you iterate through the cache for any keys prefixed with https://api.airtable.com/v0/appXYZ/Users/ if you updated the Users table? The cache bust represented here will currently only clear it for https://api.airtable.com/v0/appXYZ/Users rather than https://api.airtable.com/v0/appXYZ/Users/*, which is maybe what a user might expect.

Reply:
The current caching strategy addresses the scenario you listed out in steps 1 - 3.  So it's currently safe to cache the GET requests with the `recABC` portion. It should also address the `filterByFormula` scenario. 

We can guarantee this because `createClearKey` (in utils/cache.js) generates `version/baseId/tableIdOrName` as the string/key used to filter out the keys that will be deleted.